src/main/java/dev/openfga/autoconfigure/OpenFgaProperties.java

```diff
@@ -29,40 +31,265 @@ public class OpenFgaProperties implements InitializingBean {
      */
     private Credentials credentials;
 
+    /**
+     * The HTTP user agent header to use for requests, default openfga-sdk java/{version}
+     */
+    private String userAgent;
+
+    /**
+     * The read timeout for requests, default 10 seconds
+     */
+    private Duration readTimeout;
```

Comment:
Perhaps for the `Duration` types, we could use the `@DurationUnit` [annotation](https://docs.spring.io/spring-boot/api/java/org/springframework/boot/convert/DurationUnit.html) and set it to [ChronoUnit.SECONDS](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/temporal/ChronoUnit.html#SECONDS)?